dradis.py

```diff
@@ -965,31 +965,55 @@ def get_attachment(self, project_id: int, node_id: int, filename: int):
 
         return result
 
-    def create_attachment(self, project_id: int, node_id: int):
-        """Create a new attachment
+    def _create_multipart(self, endpoint, project_id, files):
+        """Generic function to create for an endpoint using multipart POST (attachments)"""
 
-        :param project_id: ID for the project
-        :param node_id: ID for the node to create attachment for
-        """
+        # BUILD API ENDPOINT
+        url = self.__url + endpoint
 
-        raise NotImplementedError()
+        # HTTP REQUEST TYPE
+        req_type = "POST"
 
-        # Add required header to set
+        # SET HEADERS
         self._add_project_header(project_id)
+        
+        # Remove  application/json Content-type and let requests handle the multipart/format-data type + boundary automatically
+        self.__headers.pop('Content-type', None)
```

Comment:
the __headers variable is mutated here, but it is used in other requests. So, other requests will not have the 'Content-Type' header set after _create_multipart is run. Make a new headers object with only the correct headers:

```
headers = {k:v for k,v in self.__headers.items() if k != 'Content-Type'}
```

Reply:
Good point, thanks for the feedback.  I updated the code and took in accounts your comments.